orangecontrib/text/widgets/owconcordance.py

```diff
@@ -0,0 +1,265 @@
+        self.conc_view.setWordWrap(False)
+        self.conc_view.setSelectionBehavior(QTableView.SelectRows)
+        self.conc_view.setSelectionModel(DocumentSelectionModel(self.model))
+        # connect selectionChanged to self.commit(), which will be
+        # updated by gui.auto_commit()
+        self.conc_view.selectionModel().selectionChanged.connect(lambda:
+                                                                 self.commit())
+        self.conc_view.horizontalHeader().hide()
+        self.conc_view.setShowGrid(False)
+        self.mainArea.layout().addWidget(self.conc_view)
+        self.set_width()
+
+        # Auto-commit box
+        gui.auto_commit(self.controlArea, self, 'autocommit', 'Commit',
+                        'Auto commit is on')
+
+    def sizeHint(self):
+        return QSize(600, 400)
+
+    def set_width(self):
+        self.model.set_width(self.width)
+
+    def set_data(self, data=None):
+        self.corpus = data
+        if data is not None and not isinstance(data, Corpus):
+            self.corpus = Corpus.from_table(data.domain, data)
+        self.model.set_data(self.corpus)
+        self.update_info()
+        self.refresh_search()
+        self.commit()
+
+    def refresh_search(self):
+        self.model.set_word(self.word)
+        if self.corpus is not None and self.word:
+            self.update_info()
+        self.conc_view.resizeColumnToContents(1)
+        self.resize_columns()
+        self.conc_view.resizeRowsToContents()
+
+    def resize_columns(self):
+        col_width = (self.conc_view.width() -
+                     self.conc_view.columnWidth(1)) / 2 - 12
+        self.conc_view.setColumnWidth(0, col_width)
+        self.conc_view.setColumnWidth(2, col_width)
+
+    def resizeEvent(self, event):
+        super().resizeEvent(event)
+        self.resize_columns()
+
+    def update_info(self):
+        if self.corpus is not None:
+            self.n_documents = len(self.corpus)
+            self.n_matching = '{}/{}'.format(self.model.matching_docs(),
+                                             self.n_documents)
+            self.n_tokens = sum(map(len, self.corpus.tokens)) \
+                if self.corpus.has_tokens() else 'n/a'
+            self.n_types = len(self.corpus.dictionary) \
+                if self.corpus.has_tokens() else 'n/a'
+        else:
+            self.n_documents = ''
+            self.n_matching = ''
+            self.n_tokens = ''
+            self.n_types = ''
+
+    def commit(self):
+        rows = [sel_range.top() for sel_range
+                in self.conc_view.selectionModel().selection()]
+        selected_docs = sorted(set(self.model.word_index[row][0]
+                                   for row in rows))
+        if selected_docs:
+            selected = self.corpus[selected_docs]
+            self.send("Selected Documents", selected)
+        else:
+            self.send("Selected Documents", None)
+
+
+if __name__ == '__main__':
```

Comment:
Since you will have to make some other changes anyway, you can put an extra space between `:` and `#`. PEP8 recommends [(at least) two spaces between statements and comments](https://www.python.org/dev/peps/pep-0008/#inline-comments). Also at the other two places with the pragma comment. Pylint and PyCharm will appreciate it.